File: 2023/01.pl
#!/usr/bin/env perl

use strict;
use warnings;
use integer;

my $ans1 = 0;
my $ans2 = 0;

my $re1 = qr/\d/;
my $re2 = qr/one|two|three|four|five|six|seven|eight|nine|\d/;
my %tr = (
	one => 1,
	two => 2,
	three => 3,
	four => 4,
	five => 5,
	six => 6,
	seven => 7,
	eight => 8,
	nine => 9,
);

while (<>) {
	my $d1;
	my $d2;

	$d1 = $1 if (/($re1)/);
	$d2 = $1 if (/.*($re1)/);
	$ans1 += $d1 . $d2;

	$d1 = $tr{$1} || $1 if (/($re2)/);
	$d2 = $tr{$1} || $1 if (/.*($re2)/);
	$ans2 += $d1 . $d2;
}

print "$ans1 $ans2\n";

File: 2025/01.c
#include <stdio.h>

int main(void)
{
	long ans1 = 0;
	long ans2 = 0;

	int x = 50;
	char d;
	int n;
	while (scanf(" %[LR]%d ", &d, &n) == 2) {
		int dx = d == 'R' ? 1 : -1;
		while (n--) {
			x += dx;
			if (x < 0) x += 100;
			else if (x >= 100) x -= 100;
			if (x == 0) ans2++;
		}
		if (x == 0) ans1++;
	}

	printf("%ld %ld\n", ans1, ans2);
	return 0;
}
